### PR TITLE
update embedded libclang to 13.0.1

### PR DESCRIPTION
**Pull request by kevinushey**
### Intent

Addresses https://github.com/rstudio/rstudio/issues/7554.

### Approach

Update the embedded version of libclang on Windows to 13.0.1, and also make a number of tweaks to ensure it behaves well with rtools42.

### Automated Tests

N/A

### QA Notes

Test via notes in https://github.com/rstudio/rstudio/issues/7554.

### Checklist

- [x] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [x] This PR passes all local unit tests

### CODE REVIEW

File: src/cpp/session/session-options.json [hunk 1]
```diff
@@ -719,13 +719,6 @@
             "defaultValue": {"code": "kDefaultRsclangPath", "description": "bin/rsclang"},
             "description": "Specifies the path to the libclang shared library"
          },
-         {
-            "name": "external-libclang-headers-path",
```
Comment (gtritchie) [inline src/cpp/session/session-options.json:723]: Will this cause problems (like session failing to start) if someone had this setting in rsession.conf? Maybe we should leave this in and mark it as `isDeprecated`, such as seen here?

https://github.com/rstudio/rstudio/blob/7781d230ce5abdcd20d185b385f1a0ee01df6da8/src/cpp/session/session-options.json#L575-L581

**Review by gtritchie — CHANGES_REQUESTED** [review]

**Review by gtritchie — COMMENTED** [review]
We should also mention this in the NEWS, I think.

File: src/cpp/session/session-options.json (hunk 1 above)
Comment (kevinushey) [reply src/cpp/session/session-options.json:723]: Good point; I've added it back in. Better safe than sorry.

**Review by gtritchie — APPROVED** [review]
LGTM!

**Pull request merged**